docker/pypi/wmagent/init.sh

```diff
@@ -331,6 +341,29 @@ EOF
     echo "-----------------------------------------------------------------------"
 }
 
+check_wmatag() {
+    # A function to check if there have been any change in the WMA_TAG since last
+    # WMAgent initialization at the host. It does the check by comparing the whole
+    # WMA_TAG from the environment and the one previously recorded at relational
+    # database in the table `wma_init`.
+    #
+    # NOTE: Before we even start checking all the rest of the initialization flags, we should
+    #       check if this agent has ever been used, and if there is any patch version
+    #       or release candidate change between the current one and the one previously
+    #       initialized at the host. If such discrepancy is confirmed, we should enforce Runtime
+    #       code copy by deleting only the $wmaInitSqlDB and let the rest of the init process take over.
+    _init_valid $wmaInitSqlDB && {
+        echo "$FUNCNAME: This agent has been previously initialized. Checking for WMAgent version change since last run."
+        local wmaTagCurr=$WMA_TAG
+        local wmaTagSqlDB=$(_exec_sql "select init_value from wma_init where init_param = 'wma_tag';")
+        [[ $wmaTagCurr == $wmaTagSqlDB ]] || {
```

Comment:
Shouldn't it be `!=`?

Reply:
No, It must be `==` and we should act on broken condition i.e. with `||` logical operator. Or... if we change it 
to `!=`  to include the negative inside the condition itself,  then we should act on satisfied (negative) condition with `&&` ... And you know my opinion about double negation in the logic.....

Reply:
Ahh, I am always getting confused with those. It looks like my brain has a hard time to read this without an explicit if :) Thanks Todor.